test/libsolidity/syntaxTests/inlineAssembly/shadowing/function.sol

```diff
@@ -0,0 +1,8 @@
+contract C {
+    function f() public pure {
+        assembly {
+            let f := 1
```

Comment:
Should we go ahead and just disallow *all* collisions, including stuff that can't (yet) be referenced like contracts or functions?

Reply:
If that is easy to do, then it sounds good, I would say.

Reply:
Ok - I specifically had to prevent these from being errors, so it's actually easier to make all of them errors.